### PR TITLE
support decoding into anonymous struct fields

### PR DESCRIPTION
**Pull request by timob**
see https://github.com/naoina/toml/issues/38

### CODE REVIEW

File: util.go [hunk 1]
```diff
@@ -10,8 +10,8 @@ const fieldTagName = "toml"
 
 // fieldCache maps normalized field names to their position in a struct.
 type fieldCache struct {
-	named map[string]fieldInfo // fields with an explicit name in tag
-	auto  map[string]fieldInfo // fields with auto-assigned normalized names
+	named map[string]*fieldInfo // fields with an explicit name in tag
```
Comment (glinton) [inline util.go]: No need to use pointers. Tested PR without and it works great.

**Review by glinton — APPROVED** [review]

File: util.go (hunk 1 above)
Comment (timob) [reply util.go]: i've added you to the timob/toml collaborators, can you push the changes to timob:master ?

**Comment by timob** [comment]
@glinton i don't think that commit completely does what is needed.

I don't use TOML anymore so if probably will close this PR.

**Comment by glinton** [comment]
> i don't think that commit completely does what is needed.

What more do you think it needs?

File: util.go [hunk 2]
```diff
@@ -22,20 +22,41 @@ type fieldInfo struct {
...
+				descend(append(index, ft.Index...), ft.Type)
+				continue
+			}
+
+			info := fieldInfo{index: append(index, ft.Index...), name: ft.Name, ignored: col == "-"}
+			if col == "" || col == "-" {
+				name := cfg.NormFieldName(rt, ft.Name)
+				// discard any duplicate names
+				if v, ok := auto[name]; ok {
+					v.ignored = true
```
Comment (timob) [inline util.go]: what is this for?

Comment (glinton) [reply util.go]: good point.. will update

**Pull request closed without merging**